### PR TITLE
[WIP] Refactor adjoint metric tensor

### PR DESCRIPTION
**Pull request by comp-phys-marc**
**Context:** We would like to update `adjiont_metric_tensor` to work without `tape_expand_fn` since we are deprecating `tape_expand_fn`.

**Description of the Change:** So far, a messy prototype of how we could get things working without `tape_expand_fn`.

**Benefits:** The transform doesn't depend on deprecated internals.

**Possible Drawbacks:** This is a super messy prototype work-in-progress.

**Related ShortCut Stories:** [sc-107783]

### CODE REVIEW

**Comment by comp-phys-marc** [comment]
Just looking for some pre-review comments as I am a bit stumped with the remaining errors.

File: pennylane/gradients/adjoint_metric_tensor.py [hunk 1]
```diff
@@ -60,26 +62,56 @@ def _group_operations(tape):
...
+        stopping_condition = _trainable_multipar_stopping_fn
+
+    else:
 
-    interface = math.get_interface(*tape.get_parameters())
-    use_tape_argnum = interface == "jax"
-    expand_fn = create_expand_trainable_multipar(tape, use_tape_argnum=use_tape_argnum)
-    return [expand_fn(tape)], lambda x: x[0]
+        def _argnum_trainable_multipar(obj):
+            return _multipar_stopping_fn(obj) or (not math.requires_grad(obj) and obj.has_generator)
```
Comment (JerryChen97) [inline pennylane/gradients/adjoint_metric_tensor.py:102]: Mind double check if this new stopping condition is actually working?

Comment (comp-phys-marc) [reply pennylane/gradients/adjoint_metric_tensor.py:102]: It seems to me like this doesn't work as intended since it is not the same as before the refactor. This is because:

```
>>> (obj not in trainable_ops) == (not math.requires_grad(obj))
False
```

More evidence to support this is that when I don't add the check for `obj.has_generator` (which is my addition here), we end up with `ops` that don't have generators and therefore can't be processed by the `processing_fn`.

However, I do not know what I could do to fix this.

Comment (JerryChen97) [reply pennylane/gradients/adjoint_metric_tensor.py:102]: Oh... then no wonder why those jax tests failed; highly likely we have to explore more to see how this can get around.
Would this block the ongoing deprecation in any way?

Comment (comp-phys-marc) [reply pennylane/gradients/adjoint_metric_tensor.py:102]: Yes, this unfortunately blocks all three of my deprecations.

Comment (comp-phys-marc) [reply pennylane/gradients/adjoint_metric_tensor.py:102]: @albi3ro might have a fix here: https://github.com/PennyLaneAI/pennylane/pull/8960

**Comment by comp-phys-marc** [comment]
Replace by https://github.com/PennyLaneAI/pennylane/pull/8960
